src/wallet/walletdb.h

```diff
@@ -157,6 +157,9 @@ class WalletBatch
             return false;
         }
         m_database.IncrementUpdateCounter();
+        if (m_database.nUpdateCounter % 1000 == 0) {
```

Comment:
I get that this makes the code simple, but IMO this defeats the batch purpose. Like, it shouldn't be something to worry about here.

Reply:
I don't understand what you mean by 'defeats the batch purpose'. Are there cases where the caller is depending on the batch not being flushed?

I think it is far better engineering practice to deal with periodic flushing once, inside the batch-management code, than to deal with it independently at each callsite. If there's some situation where periodic flushing is undesirable, I think it would be better to offer both versions.

Reply:
@gwillen I mean you should be able to create a batch with 1001 changes no? I was suggesting something on top of `WalletBatch` to do that.

Reply:
Well, you can add as many changes as you want, it just flushes after every 1000. This should be completely transparent to the user. Unless there is some reason that flushing can hurt, like the caller is using the batch as though it were an atomic transaction?

Reply:
> Unless there is some reason that flushing can hurt, like the caller is using the batch as though it were an atomic transaction?

There is only one place where an atomic transaction is used, and if an atomic transaction is active, then `Flush()` will do nothing.

Reply:
How about add an optional argument to `WalletBatch` that indicates the # of updates to flush after? This makes the operation opt-in/explicit and also allows for optionally adjusting the flushes to the number of operations - e.g. 2x600 rather than 1000 and 200, if that's desirable.

Reply:
Well I may be wrong but I see the batch as an atomic transaction. Making 1500 updates and then aborting the batch should not commit 1000 updates IMO.

Reply:
> Well I may be wrong but I see the batch as an atomic transaction. Making 1500 updates and then aborting the batch should not commit 1000 updates IMO.

Except that's not how the batch works normally. Batches are not atomic unless you explicitly make them so by starting a transaction with the underlying `BerkeleyBatch`.  In that case, then the flushing won't do anything. In the normal case, each update in a batch itself is atomic and should be expected to be written to the wallet file. There is no aborting. As soon as a batch goes out of scope, every single update is committed to the file. If you make 1500 updates, the first 1000 will be written after you do the 1000th write operation, and then the last 500 will be written when the batch object is deleted.

Reply:
I believe that is out of date

Reply:
I've updated the comment.